src/core/xref.js

```diff
@@ -63,13 +64,32 @@ class XRef {
     // stream.
     if (this._newTemporaryRefNum === null) {
       this._newTemporaryRefNum = this.entries.length || 1;
+      if (this._newPersistentRefNum) {
+        this._persistentRefsCache = new Map();
+        for (
+          let i = this._newTemporaryRefNum;
+          i < this._newPersistentRefNum;
+          i++
+        ) {
+          // We clean the cache to avoid any conflict with the refs created
+          // during saving.
```

Comment:
Can we please clarify here that this only temporary, since that may be helpful to a future reader of the code?
```suggestion
          // We *temporarily* clear the cache, see `resetNewTemporaryRef` below,
          // to avoid any conflict with the refs created during saving.
```